src/main/java/org/zalando/nakadi/service/EventStream.java

```diff
@@ -47,7 +47,7 @@ public EventStream(final EventConsumer eventConsumer,
         this.writer = writer;
     }
 
-    public void streamEvents(final AtomicBoolean connectionReady) {
+    public void streamEvents(final AtomicBoolean connectionReady, final Runnable checkAuthorization) {
```

Comment:
I think to make it more consistent it would make sense to pass here a second atomic boolean instead of runnable. And you eve have that atomic boolean already in controller code.